### ISSUE TITLE
Removed backtick around property named "self"

### ISSUE THREAD

**Issue by guillaumealgis** [nicklockwood/SwiftFormat#1350]
First, let me agree on the fact that naming a property `self` in Swift is a bad idea ©.

However, I encountered a real world API where one of the fields of the objects I'm getting have a property named `self`. So I implemented my Codable model like so:

```swift
import Foundation

struct Fruit: Codable {
    var id: Int
    var name: String
    let `self`: URL

    func printURL() {
        print("My URL is \(self.`self`)")
    }
}

let json = """
{
    "id": 42,
    "name": "Banana",
    "self": "http://api.example.com/fruits/42"
}
"""

let decoder = JSONDecoder()
let fruit = try decoder.decode(Fruit.self, from: Data(json.utf8))
fruit.printURL()  // My URL is http://api.example.com/fruits/42
```

But when running SwiftFormat on this file, it removes the backticks around `` `self` `` inside the function:

```diff
<         print("My URL is \(self.`self`)")
---
>         print("My URL is \(self.self)")
```

This obviously makes the code incorrect, as the `printURL()` function is no longer printing the `` `self` `` property but the `Fruit` instance itself.

Note that I can work around the issue by implementing codable manually etc. But that still is a bug of SwiftFormat to change the meaning of the code :)

---------------

_SwiftFormat 0.50.7_

**Comment by nicklockwood** [comment]
You're right, this is a bug

**Comment by nicklockwood** [comment]
@guillaumealgis fixed in 0.50.8

**Comment by guillaumealgis** [comment]
Thanks a lot for the quick fix @nicklockwood !

I can confirm this is fixed in my real-world use case :)
